diff -ru xpenguins-2.2.new/src/toon_signal.c xpenguins-2.2/src/toon_signal.c
--- xpenguins-2.2.new/src/toon_signal.c	2001-04-25 01:33:43.000000000 +0200
+++ xpenguins-2.2/src/toon_signal.c	2026-09-09 21:07:34.740267312 +0200
@@ -16,6 +16,7 @@
  *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 #include <signal.h>
+#include <stdlib.h>
 #include "toon.h"
 
 /* SIGNAL AND ERROR HANDLING FUNCTIONS */
