diff -u xpenguins-2.2.new/src/toon.h xpenguins-2.2/src/toon.h
--- xpenguins-2.2.new/src/toon.h	2001-09-30 20:21:45.000000000 +0200
+++ xpenguins-2.2/src/toon.h	2026-09-09 21:04:00.901045471 +0200
@@ -118,7 +118,7 @@
   unsigned int wid;
   XRectangle pos;
 } __ToonWindowData;
-typedef int __ToonErrorHandler();
+typedef int __ToonErrorHandler(Display *toon_display, XErrorEvent *error);
 void __ToonSignalHandler(int sig);
 int __ToonError(Display *toon_display, XErrorEvent *error);
 void __ToonExitGracefully(int sig);
